Gems/PhysX/Code/Source/Scene/PhysXScene.cpp

```diff
@@ -1348,5 +1361,42 @@ namespace PhysX
         AZStd::for_each(m_packedIndices.begin(), m_packedIndices.end(), applyFunction);
     }
 
+    void PhysXScene::QueuedActiveBodyIndices::ApplyParallel(const AZStd::function<void(AzPhysics::SimulatedBodyIndex)>& applyFunction, physx::PxScene* pxScene)
+    {
+        AZ::TaskGraph taskGraph("Parallel Sync");
+        AZ::TaskGraphEvent finishEvent("Parallel sync event");
+
+        {
+            AZ_PROFILE_SCOPE(Physics, "Sync Setup");
+
+            size_t batchSize = physx_parallelTransformSyncBatchSize;
+            size_t fullSize = m_packedIndices.size();
+            for (size_t i = 0; i < fullSize; i += batchSize)
+            {
+                AZ::TaskDescriptor taskDescriptor{"SyncTask", "Physics"};
+                taskGraph.AddTask(
+                    taskDescriptor,
+                    [start = i, end = AZStd::min(i + batchSize, fullSize), &applyFunction, pxScene, this]()
+                    {
+                        AZ_PROFILE_SCOPE(Physics, "Sync Task");
+
+                        // Note: It is important to keep the scene locked for read for the entire task execution.
+                        // Otherwise the functions reading data from the rigid body will have to lock it locally.
+                        // This causes a huge amount of context switches making the execution of each task ~20x slower. 
+                        PHYSX_SCENE_READ_LOCK(pxScene);
```

Comment:
question: I'm just curious about how this works... If we lock the `pxScene` for each task, will that mean other tasks will have to wait and run serially? Or if they're all readers, is it fine for them to all hold this lock at the same time so they can all run in parallel? Thanks!

Reply:
All readers can read simultaneously and nested read-locks cost nothing. 
It's the first read lock per thread is expensive, so we do this one higher level read-lock to ensure all further read-locks nested.

Reply:
I see, that's great thanks for clarifying :+1: